sdk/src/callback_signer.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright 2024 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+#![deny(missing_docs)]
+
+//! The `callback_signer` module provides a way to obtain a [`Signer`]
+//! using a callback and public signing certificates.
+
+use crate::{
+    error::{Error, Result},
+    Signer, SigningAlg,
+};
+
+/// Defines a callback function interface for a signer.
+///
+/// The callback should return a signature for the given data.
+/// The signature should be in the format expected by the `Signer`.
+/// The callback should return an error if the data cannot be signed.
+pub type CallbackFunc = dyn Fn(*const (), &[u8]) -> std::result::Result<Vec<u8>, Error>;
+
+/// Defines a signer that uses a callback to sign data.
+///
+/// The private key should only be known by the callback.
+/// This structure is private to this module.
```

Comment:
If this is true, you could remove this comment and use `pub(crate)` or `pub(super)` so that the compiler will enforce that for you.

Reply:
The comment is wrong, this is now part of the public interface. I'll remove the comment.

Reply:
Those comments are out of date. The intention is to have this be a standard part of the API. The various bindings depend on this, so it makes sense to share it here.